sql/src/main/java/io/crate/execution/engine/distribution/TransportDistributedResultAction.java

```diff
@@ -157,6 +189,23 @@ void pushResult(String node, DistributedResultRequest request, ActionListener<Di
             scheduler.schedule(operationRunnable::run, delay.getMillis(), TimeUnit.MILLISECONDS);
             return operationRunnable;
         } else {
+            List<String> excludedNodeIds = Collections.singletonList(clusterService.localNode().getId());
+            /* The upstream (DistributingConsumer) forwards failures to other downstreams and eventually considers its job done.
+             * But it cannot inform the handler-merge about a failure because the JobResponse is sent eagerly.
```

Comment:
what do you mean by "eagerly"? I assume you mean the response was sent _before_ the upstream DistributingConsumer actually gets the job done?

Reply:
yep, that's what I meant with eagerly